pyannote/audio/pipelines/speaker_diarization.py

```diff
@@ -530,20 +538,35 @@ def apply(
         )
         diarization.uri = file["uri"]
 
-        # when reference is available, use it to map hypothesized speakers
-        # to reference speakers (this makes later error analysis easier
-        # but does not modify the actual output of the diarization pipeline)
         if "annotation" in file and file["annotation"]:
-            return self.optimal_mapping(file["annotation"], diarization)
+            # when reference is available, use it to map hypothesized speakers
+            # to reference speakers (this makes later error analysis easier
+            # but does not modify the actual output of the diarization pipeline)
+            label_map = self.optimal_mapping(
+                file["annotation"], diarization, mapping_only=True
+            )
 
-        # when reference is not available, rename hypothesized speakers
-        # to human-readable SPEAKER_00, SPEAKER_01, ...
-        return diarization.rename_labels(
-            {
+        else:
+            # when reference is not available, rename hypothesized speakers
+            # to human-readable SPEAKER_00, SPEAKER_01, ...
+            label_map = {
                 label: expected_label
                 for label, expected_label in zip(diarization.labels(), self.classes())
             }
-        )
+
+        diarization = diarization.rename_labels(label_map)
+
+        if not return_embeddings:
+            return diarization
+        else:
+            embedding_map = {}
+            for k, label in label_map.items():
+                if k < centroids.shape[0]:
+                    embedding_map[label] = centroids[k]
+            return {
+                "diarization": diarization,
+                "speaker_embeddings": embedding_map,
+            }
```

Comment:
I'd rather return a tuple (_à la_ [`np.unique`](https://numpy.org/doc/stable/reference/generated/numpy.unique.html))

```suggestion
            return diarization, embedding_map
```